### ISSUE TITLE
Streaming virtual disc images

### ISSUE THREAD

**Issue by kiler129** [pikvm/pikvm#101]
**Is your feature request related to a problem? Please describe.**
One of the best features I use on systems like iLO is the ability to just mount an ISO image and reinstall the server. While booting an image from the PiKVM host is supported this solution poses problems when networks are configured to be secure.

Usually it's recommended that the management network is separated, at least on the switch level with a dedicated port. Currently PiKVM is a perfect solution for that. However, usually management networks are isolated and with no direct access to the Internet.

This configuration forces me to upload the image to PiKVM host through a VPN to mount it. It's a mild inconvenience.

**Describe the solution you'd like**
There should be an option to stream the ISO image through the browser:

![Screen Shot 2020-10-25 at 12 08 39 AM](https://user-images.githubusercontent.com/1227834/97099409-99e74080-1656-11eb-895e-d6adfadbfe4c.png)
![Screen Shot 2020-10-25 at 12 10 25 AM](https://user-images.githubusercontent.com/1227834/97099411-9e135e00-1656-11eb-8a4e-da6efbf6bd5f.png)

**Describe alternatives you've considered**
Uploading the image to PiKVM & mounting it.

**Additional context**
- Additionally, making PiKVM datastore read-only could make it more reliable for long term use. With the requirement of r/w storage for images the fs cannot be made read-only.
- iLO serves ISO images via WebSocket. If there will be an interest I can poke around for more details.

WDYT?

**Comment by mdevaev** [comment]
This is a good idea, but at the moment it is associated with some difficulties, local mounting of images is limitation of the kernel. However, it would be nice to do it somehow.

BTW the Pi-KVM file system is read-only. Images are uploaded on a separate partition (not rootfs) that is also permanently read-only and only remounted to read-write while uploading.

**Comment by kiler129** [comment]
> This is a good idea, but at the moment it is associated with some difficulties, local mounting of images is limitation of the kernel. 

Do you know about any limitations as of know, or it just wasn’t expired previously?

I’m asking since I can poke around but if someone already tried and stepped into some issues it will be nice to not repeat the same steps :)

**Comment by mdevaev** [comment]
No, no one has tried it. The kernel just requires specifying a local image, that's all. Maybe we can create a fuse driver and it will swallow it, but I haven't tried it.

**Comment by mdevaev** [comment]
some information so as not to forget: https://wicg.github.io/file-system-access
